gen/generate.go

```diff
@@ -309,22 +285,28 @@ func generateModule(m *compile.Module, i thriftPackageImporter, builder *generat
 			if _, err := builder.AddRootService(service); err != nil {
 				return nil, err
 			}
+		}
 
-			serviceFiles, err := Service(g, service)
-			if err != nil {
-				return nil, fmt.Errorf(
-					"could not generate code for service %q: %v",
-					serviceName, err)
-			}
-
-			if !o.NoServiceHelpers {
-				for name, buff := range serviceFiles {
-					files[name] = buff.Bytes()
-				}
-			}
+		if err = Services(g, m.Services); err != nil {
+			return nil, fmt.Errorf(
+				"could not generate code for services %v", err)
 		}
 	}
 
+	buff := new(bytes.Buffer)
+	if err := g.Write(buff, nil); err != nil {
+		return nil, fmt.Errorf(
+			"could not write file %q: %v", o.OutputFile, err)
+	}
+
+	// The default is the package name
+	name := packageName + ".go"
+	if len(o.OutputFile) > 0 {
+		name = o.OutputFile
```

Comment:
Name must be calculated before the error message thrown on line 299.

Reply:
Adjusted